excise/tasks.py

```diff
@@ -51,49 +51,65 @@ def api_post_request_task(
         span.set_tag(opentracing.tags.COMPONENT, "tax")
         span.set_tag("service.name", "avatax_excise")
 
-        response = api_post_request(transaction_url, data, config)
-    msg = f"Order sent to Avatax Excise. Order ID: {order.token}"
-    if not response or "Error" in response.get("Status"):
-        errors = response.get("TransactionErrors", [])
-        avatax_msg = ""
-        for error in errors:
-            avatax_msg += error.get("ErrorMessage", "")
-        msg = f"Unable to send order to Avatax Excise. {avatax_msg}"
+        tax_response = api_post_request(transaction_url, data, config)
+
+    msg = ""
+    if not tax_response:
+        msg = (
+            "Empty response received from Excise API, "
+            f"Order: {order.token}"
+        )
+        logger.warning(msg)
+        external_notification_event(
+            order=order, user=None, message=msg, parameters=None
+        )
+        return
+    elif tax_response.get("ReturnCode", -1) != 0:
+        errors = tax_response.get("TransactionErrors", [])
+        error_msg = ". ".join([
+            error.get("ErrorMessage", "")
+            for error in errors
+        ])
+        msg = f"Unable to send order to Avatax Excise. {error_msg}"
         logger.warning(
             "Unable to send order %s to Avatax Excise. Response %s",
             order.token,
-            response,
+            tax_response,
         )
-        raise TaxError(msg)
+        external_notification_event(
+            order=order, user=None, message=msg, parameters=None
+        )
+        return
     else:
-        user_tran_id = response.get('UserTranId')
+        msg = f"Order sent to Avatax Excise. Order ID: {order.token}"
+        user_tran_id = tax_response.get("UserTranId")
         if config.autocommit and commit_url and user_tran_id:
             commit_url = commit_url.format(user_tran_id)
-            response = api_post_request(
+            commit_response = api_post_request(
                 commit_url,
                 {},
                 config,
             )
-            errors = response.get("TransactionErrors", [])
-            avatax_msg = ""
-            for error in errors:
-                avatax_msg += error.get("ErrorMessage", "")
             msg = f"Order committed to Avatax Excise. Order ID: {order.token}"
-            if not response or "Error" in response.get("Status"):
-                msg = f"Unable to commit order to Avatax Excise. {avatax_msg}"
+            if not commit_response or "Error" in commit_response.get("Status"):
```

Comment:
If `commit_response` will not have `Status`, get will return default value == `None` , then `"Error" in None` will raise `TypeError`.  I would set empty dict as a default value.

Reply:
I think I get your point, but response status should be a string instead of a dictionary, so a empty string might work better here.

Fixed here: https://github.com/saleor/saleor-plugin-avatax-excise/pull/11/commits/cbd47fdc24fc59eb16a7eb680a5ceab94fa49641

Thanks!